src/engine/input/InputManager.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Godot;
+
+/// <summary>
+///   A handler for inputs. Listens for inputs and notifies the input attributes.
+/// </summary>
+/// <remarks>
+///   <para>
+///     This is an AutoLoad class.
+///   </para>
+/// </remarks>
+public class InputManager : Node
+{
+    private static InputManager staticInstance;
+
+    /// <summary>
+    ///   A list of all loaded attributes
+    /// </summary>
+    private readonly List<InputAttribute> attributes = new List<InputAttribute>();
```

Comment:
I think a good way to change this would be to change this into a dictionary like: `Dictionary<InputAttribute, List<WeakReference>>` in order to store the instances related to each input attribute. That way this class owns the instance references so this doesn't have to violate encapsulation by directly accessing InputAttribute.Instances.